### ISSUE TITLE
serializeToString exception with xmldom@0.1.20

### ISSUE THREAD

**Issue by benbracha** [jindw/xmldom#146]
Since this morning, we started to get errors using grunt-svg-sprite that is using (indirectly, by svg-sprite) xmldom package.

Things seems to work with xmldom@0.1.19 but not on xmldom@0.1.20.

This is the exception:

```
TypeError: Object <svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 24 24" enable-background="new 0 0 24 24" xml:space="preserve">
<path fill="#808080" d="M7,19.8c0.5,0.3,1,0.6,1.6,0.8L8.7,23c2.1,1.2,5.7,0.3,5.7,0.3l0.6-2.5l2.3-1.7l2.4,1.2
	c2.1-1.2,3.1-4.8,3.1-4.8l-2-1.9l0.5-2.8l2.1-1.6c0,0-0.7-3.6-2.7-5l-2.2,0.9C18,4.7,17.6,4.3,17.1,4c-0.4-0.2-0.8-0.5-1.2-0.6
	l-0.5-2.5c0,0-3.6-1.1-5.7,0L9.5,2.9C8.4,3.2,7.3,3.8,6.4,4.5l-2-1.1C2.3,4.6,1.2,8.1,1.2,8.1l1.8,1.7c-0.3,1.2-0.3,2.3-0.1,3.5
	L1,14.4c0,2.4,2.4,5.1,2.4,5.1L5.9,19C6.3,19.2,6.6,19.5,7,19.8z M8.4,9.6c1.3-2,3.9-2.6,5.9-1.3c2,1.3,2.6,3.9,1.3,5.9
	c-1.3,2-3.9,2.6-5.9,1.3C7.8,14.2,7.2,11.6,8.4,9.6z"/>
</svg> has no method 'serializeToString'
  at XMLSerializer.serializeToString (/home/jenkins/workspace/ui_general_builder/node_modules/svg-sprite/node_modules/xmldom/dom.js:906:7)
  at Object.SVGShape.getSVG (/home/jenkins/workspace/ui_general_builder/node_modules/svg-sprite/lib/svg-sprite/shape.js:278:33)
  at Object.module.exports [as svgo] (/home/jenkins/workspace/ui_general_builder/node_modules/svg-sprite/lib/svg-sprite/transform/svgo.js:63:24)
  at /home/jenkins/workspace/ui_general_builder/node_modules/svg-sprite/lib/svg-sprite.js:168:38
  at fn (/home/jenkins/workspace/ui_general_builder/node_modules/svg-sprite/node_modules/async/lib/async.js:638:34)
  at Object._onImmediate (/home/jenkins/workspace/ui_general_builder/node_modules/svg-sprite/node_modules/async/lib/async.js:554:34)
  at processImmediate [as _immediateCallback] (timers.js:330:15)
```

Any idea?
(@jkphl FYI)

**Comment by jkphl** [comment]
@benbracha Ooops! Please let me know if it turns out that this is an *svg-sprite* problem rather than an *xmldom* problem ... Thanks!

**Comment by benbracha** [comment]
I does work with the previous xmldom version, so I assume it is not svg-sprite issue.

**Comment by jkphl** [comment]
Right, there must've been a change in *xmldom* then. But it's still possible that I've made some stupid mistake in *svg-sprite* and e.g. used a unintended feature of *xmldom* that never should've been there and that has been "fixed" now ... Looking forward to a solution either way! :)

**Comment by moshezvi** [comment]
+1 on this issue, confirmed that it's an 0.1.20 issue (after downgrading to 0.1.19 it works).

**Comment by noahgrant** [comment]
+1, confirmed it works after downgrading (with gulp-svg-sprite).

**Comment by mikeobrien** [comment]
+1

**Comment by benbracha** [comment]
As a general note, as this is not the first time we got failures due to components change (although we use strict dependencies versions - without ^, ~ etc.).
Maybe, as a practice, components should use strict versions on their dependencies?
Otherwise, you cannot be sure that 2 builds of your application will result the same - and we already saw in the past (not specifically to this component) such issues. This makes CI and deployment plans risky.

I would be happy to hear your thoughts / open this thread wherever you think appropriate.

**Comment by jkphl** [comment]
@benbracha Not sure who exactly you were talking to, but I'll consider this in my next releases. ;)

**Comment by benbracha** [comment]
It was a general comment, just to hear some thoughts.
But yes - @jkphl  it worth to consider it on grunt-svg-sprite and svg-sprite and get more stable dependencies :+1:

**Comment by johanneswuerbach** [comment]
Should be fixed by https://github.com/jindw/xmldom/pull/148

**Comment by omggga** [comment]
Same problem for me with gulp-xmlpoke@0.2.0, wich uses the latest version of xmldom as dependence.

**Comment by bp323** [comment]
Confirmed, this is currently breaking all our builds. Can the latest NPM module be reverted to 0.1.19 for now?

**Comment by lopesc** [comment]
+1, affected by this. Can you publish a version of SVG-sprite that uses the old version of xmldom?

**Comment by jkphl** [comment]
Hi all, I totally understand your situation and am already working on a fix, it will hopefully land in a couple of minutes. Please be patient, there's no need adding +1 responses. Thanks. :)

**Comment by azatoth** [comment]
I assume this issue is what caused my issue at https://github.com/highsource/jsonix/issues/134

**Comment by fetis** [comment]
Also got this issue.

As temp workaround you can go to `svg-sprite` (if you're using `gulp-svgs-sprite` as me this will be `gulp-svg-sprite/node_modules/svg-srpite`)  directory and execute
```
npm i xmldom@0.1.19
```
this will install previous version

**Comment by jkphl** [comment]
Just published new versions of *svg-sprite*, *grunt-svg-sprite* and *gulp-svg-sprite* locking the dependency version to 0.1.19

**Comment by Keysox** [comment]
+1

**Comment by ovvn** [comment]
same, locked to 0.1.19

**Comment by renodim** [comment]
semver.org recommends changing the Major for API Changes such as adding a new field to the serializeToString.

**Comment by lopesc** [comment]
+1

**Comment by omggga** [comment]
As workaround for deploy on a remote server you can use https://docs.npmjs.com/cli/shrinkwrap

**Comment by jindw** [comment]
sorry :(
it 's my fault.
fixed in 0.1.21
